ckan/views/admin.py

```diff
@@ -143,26 +144,63 @@ class TrashView(MethodView):
     def __init__(self):
         self.deleted_packages = model.Session.query(
             model.Package).filter_by(state=model.State.DELETED)
+        self.deleted_orgs = model.Session.query(model.Group).filter_by(
+            state=model.State.DELETED, is_organization=True)
+        self.deleted_groups = model.Session.query(model.Group).filter_by(
+            state=model.State.DELETED, is_organization=False)
+
+        self.data_type = {
+            u'package': self.deleted_packages,
+            u'organization': self.deleted_orgs,
+            u'group': self.deleted_groups
+        }
 
     def get(self):
-        data = dict(deleted_packages=self.deleted_packages)
-        return base.render(u'admin/trash.html', extra_vars=data)
+        if request.params.get(u'name'):
+            ent_type = request.params.get(u'name').split(u'-')[-1]
```

Comment:
A bit of mystery here. Won't it be better if you just pass `name` as is instead of splitting it? And if you are going to change this line, try not to call `dict::get` twice.